src/yaml.cpp

```diff
@@ -104,30 +103,132 @@ estd::result<YAML::Node, estd::error> readYamlFile(std::string const & path) {
-	if (!map_a.IsMap() && !map_a.IsNull()) {
-		return YamlError{"tried to merge into a YAML node that is not a map"};
+// Uitility function to check whether a value is present in a vector.
+bool Contains(const std::vector<int> &list, int x){
+	return std::find(list.begin(), list.end(), x) != list.end();
+}
+
+// Check whether item is a single item map.
+bool singleMap(YAML::Node yaml_node){
+	if (!yaml_node.IsMap() && yaml_node.size() > 1){
+		return false;
 	}
-	if (!map_b.IsMap() && !map_b.IsNull()) {
-		return YamlError{"tried to merge from a YAML node that is not a map"};
+	return true;
+}
+// Utility function to check whether key in map_b is present in map_a.
+std::optional<std::size_t> orderedDictFind(YAML::Node map_a, std::string const & key){
+	for (std::size_t j = 0; j < map_a.size(); j++){
+			if (map_a[j][key]){
+					return j;
+				}
+			
 	}
+	
+	return std::nullopt;
+}
 
-	for (YAML::const_iterator iterator = map_b.begin(); iterator != map_b.end(); iterator++) {
-		std::string key = iterator->first.as<std::string>();
-		YAML::Node value = iterator->second;
-		if (value.IsMap() && map_a[key]) {
-			auto merged = mergeYamlNodes(map_a[key], value);
-			if (!merged) {
-				merged.error().appendTrace({key, "", YAML::NodeType::Map});
-				return merged;
+// Merge ordered dictionaries.
+YamlResult<void> mergeYamlOrderedDict(YAML::Node & map_a, YAML::Node map_b){
+	// Vector containing all the indexes of b which are present in a.
+	std::vector<int> checked;
+	std::optional<std::size_t> index;
+	bool single_map = true;
+	// Boolean to determine whether the element in b is present in a or not.
+	bool found = false;
+	for (std::size_t i = 0; i < map_b.size(); i++) {
+		found = false;
+		// Check whether the element in b is a single key-value pair.
+		if (!singleMap(map_b[i])){
+			return YamlError("Ordered dictionary should only contain single item map");
+		}
+		for (YAML::const_iterator iterator = map_b[i].begin(); iterator != map_b[i].end(); iterator++){
+			std::string key = iterator->first.as<std::string>();
+			index = orderedDictFind(map_a, key);
+			if (index && !Contains(checked, index.value())) {
+				if (!singleMap(map_a[index.value()])) {
+					return YamlError("Ordered dictionary should only contain single item map");
+				}
+				mergeYamlMaps(map_a[index.value()], map_b[i]);
+				checked.push_back(index.value());
+				found = true;
+				break;
 			}
-			continue;
 		}
-		map_a[key] = value;
+		// If element in b is not present in a, push the new element to a.
+		if (!found){
+				map_a.push_back(map_b[i]);
+		}
+	}
+	
+	return estd::in_place_valid;
+}
+
+// Merge Yaml Nodes.
+YamlResult<void> mergeYamlNodes(YAML::Node & map_a, YAML::Node map_b) {
+	if (!map_a.IsMap() && map_a.Tag() != "!ordered_dict" && !map_a.IsNull()) {
+		return YamlError{"tried to merge into a YAML node that is neither a map nor an ordered dictionary"};
+	}
+	else if (!map_b.IsMap() && map_b.Tag() != "!ordered_dict" && !map_b.IsNull()) {
```

Comment:
`map_b` shouldn't be required to have the `ordered_dict` tag, but if map_a is an ordered dict, `map_b` should be verified to be a list of single-item maps.

It probably makes more sense to first check what `map_a` is, and then check the type/structure of `map_b`, while already knowing what `map_a` is.